helix-term/src/ui/picker.rs

```diff
@@ -618,29 +634,23 @@ impl<T: Item + 'static> Component for Picker<T> {
                 self.to_end();
             }
             key!(Esc) | ctrl!('c') => {
+                // Action shouldn't matter here because no item was selected
+                (self.callback_fn)(cx, None, Action::Replace);
                 return close_fn;
             }
             alt!(Enter) => {
-                if let Some(option) = self.selection() {
```

Comment:
This isn't needed anymore since the None case is handled in the callback function